javascript/index.js

```diff
@@ -4,37 +4,166 @@
+
+    getInstruction(
+      "mashedPotatoes",
+      1,
+      (step2) => {
+        document.querySelector(
+          "#mashedPotatoes"
+        ).innerHTML += `<li>${step2}</li>`;
+
+        getInstruction(
+          "mashedPotatoes",
+          2,
+          (step3) => {
+            document.querySelector(
+              "#mashedPotatoes"
+            ).innerHTML += `<li>${step3}</li>`;
+
+            getInstruction(
+              "mashedPotatoes",
+              3,
+              (step4) => {
+                document.querySelector(
+                  "#mashedPotatoes"
+                ).innerHTML += `<li>${step4}</li>`;
+
+                getInstruction(
+                  "mashedPotatoes",
+                  4,
+                  (step5) => {
+                    document.querySelector(
+                      "#mashedPotatoes"
+                    ).innerHTML += `<li>${step5}</li>`;
+                    document.querySelector(
+                      "#mashedPotatoes"
+                    ).innerHTML += `<li>Mashed potatoes are ready!</li>`;
+                    document
+                      .querySelector("#mashedPotatoesImg")
+                      .removeAttribute("hidden");
+                  },
+                  (error) => console.log(error)
+                );
+              },
+              (error) => console.log(error)
+            );
+          },
+          (error) => console.log(error)
+        );
+      },
+      (error) => console.log(error)
+    );
+  },
+  (error) => console.log(error)
+);
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction("steak", 0).then((step1) => {
+  document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
+  obtainInstruction("steak", 1).then((step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+    obtainInstruction("steak", 2).then((step3) => {
+      document.querySelector("#steak").innerHTML += `<li>${step3}</li>`;
+      obtainInstruction("steak", 3).then((step4) => {
+        document.querySelector("#steak").innerHTML += `<li>${step4}</li>`;
+        obtainInstruction("steak", 4).then((step5) => {
+          document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
+          obtainInstruction("steak", 5).then((step6) => {
+            document.querySelector("#steak").innerHTML += `<li>${step6}</li>`;
+            obtainInstruction("steak", 6).then((step7) => {
+              document.querySelector("#steak").innerHTML += `<li>${step7}</li>`;
+              obtainInstruction("steak", 7).then((step8) => {
+                document.querySelector(
+                  "#steak"
+                ).innerHTML += `<li>${step8}</li>`;
+                document.querySelector(
+                  "#steak"
+                ).innerHTML += `<li>Stake is ready!</li>`;
+                document.querySelector("#steakImg").removeAttribute("hidden");
+              });
+            });
+          });
+        });
+      });
```

Comment:
I understand the logic but when dealing with promises using the "then()" we don't do it like the callback hell, for example:
```
new Promise(function(resolve, reject) {

  setTimeout(() => resolve(1), 1000); // (*)

}).then(function(result) { // (**)

  alert(result); // 1
  return result * 2;

}).then(function(result) { // (***)

  alert(result); // 2
  return result * 2;

}).then(function(result) {

  alert(result); // 4
  return result * 2;

});
```